packages/google_maps_flutter/google_maps_flutter_android/lib/src/serialization.dart

```diff
@@ -0,0 +1,152 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:google_maps_flutter_platform_interface/google_maps_flutter_platform_interface.dart';
+
+// These constants must match the corresponding constants in Convert.java
+String _objectsToAddKey(String name) => '${name}sToAdd';
+String _objectsToChangeKey(String name) => '${name}sToChange';
+String _objectIdsToRemoveKey(String name) => '${name}IdsToRemove';
+const String _heatmapIdKey = 'heatmapId';
+const String _heatmapDataKey = 'data';
+const String _heatmapGradientKey = 'gradient';
+const String _heatmapMaxIntensityKey = 'maxIntensity';
+const String _heatmapOpacityKey = 'opacity';
+const String _heatmapRadiusKey = 'radius';
+const String _heatmapGradientColorsKey = 'colors';
+const String _heatmapGradientStartPointsKey = 'startPoints';
+const String _heatmapGradientColorMapSizeKey = 'colorMapSize';
+
+void _addIfNonNull(Map<String, Object?> map, String fieldName, Object? value) {
+  if (value != null) {
+    map[fieldName] = value;
+  }
+}
+
+/// Serialize [MapsObjectUpdates]
+Object serializeMapsObjectUpdates<T extends MapsObject<T>>(
```

Comment:
I believe this whole method and the three `_objectsTo*` getters it uses can be removed, in every platform in the PR.